File: task_equivalents/llm_translations/gemini/base/r/data_visualization/02_distribution_boxplot.R
college <- read.csv('data/csv/College.csv')
rownames(college) <- college[, 1]
college <- college[, -1]
college$Private <- as.factor(college$Private)

college$Elite <- ifelse(college$Top10perc > 50, 'Yes', 'No')
college$Elite <- as.factor(college$Elite)

boxplot(Outstate ~ Elite, data = college)

par(mfrow = c(2, 2))
hist(college$Apps, breaks = 50, xlab = 'new applications', main = '')
hist(college$Enroll, breaks = 45, xlab = 'new enrollment', main = '')
hist(college$Expend, breaks = 30, xlab = 'Instructional expenditure per student', main = '')
hist(college$Outstate, xlab = 'Out-of-state tuition', main = '')

File: task_equivalents/llm_translations/gemini/base/python/data_visualization/02_distribution_boxplot.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

college = pd.read_csv('data/csv/College.csv', index_col=0)

college['Elite'] = np.where(college['Top10perc'] > 50, 'Yes', 'No')
college['Elite'] = college['Elite'].astype('category')

# R's plot(factor, numeric) creates a boxplot
plt.figure()
college.boxplot(column='Outstate', by='Elite', grid=False)
plt.title('')
plt.suptitle('')
plt.xlabel('Elite')
plt.ylabel('Out-of-state tuition')

# dev.new() and par(mfrow=c(2,2)) equivalent
fig, axes = plt.subplots(2, 2, figsize=(10, 8))

axes[0, 0].hist(college['Apps'], bins=50, edgecolor='black', facecolor='none')
axes[0, 0].set_xlabel('new applications')

axes[0, 1].hist(college['Enroll'], bins=45, edgecolor='black', facecolor='none')
axes[0, 1].set_xlabel('new enrollment')

axes[1, 0].hist(college['Expend'], bins=30, edgecolor='black', facecolor='none')
axes[1, 0].set_xlabel('Instructional expenditure per student')

axes[1, 1].hist(college['Outstate'], edgecolor='black', facecolor='none')
axes[1, 1].set_xlabel('Out-of-state tuition')

plt.tight_layout()
plt.show()